src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/ZipFileProcessor.java

```diff
@@ -54,13 +60,36 @@ public void process(ZipInputStream zis) throws IOException {
                     break;
                 default:
                     // contract breakage
-                    throw new NoPdfFileFoundException(containerName, zipFileName);
+                    throw new NoPdfFileFoundException(containerName, zipFileName, zipEntry.getName());
             }
         }
 
         log.info("PDFs found in {}: {}", zipFileName, pdfs.size());
     }
 
+    public void assertEnvelopeHasPdfs() {
```

Comment:
Looks like the method body has some extra logic beyond what the method suggests.
Please add javadoc.